src/components/DragonDeploy/DragonDeploy.tsx

```diff
@@ -0,0 +1,188 @@
+import { Dialog, Transition } from '@headlessui/react'
+import { useEffect, useState } from 'react'
+import { Fragment } from 'react'
+import toast from 'react-hot-toast'
+import SVG from 'react-inlinesvg'
+
+import CloseCrossIcon from '@/assets/icons/close-cross.svg'
+import { Button } from '@/components/common/buttons'
+import CostEstimatesToolTip from '@/components/CostEstimatesToolTip'
+import { type Commit, type File, getDeploymentBranchFiles, hasIndexFile, uploadFiles } from '@/lib/dragondeploy'
+import { useGlobalStore } from '@/stores/globalStore'
+import { Deployment } from '@/types/repository'
+
+export default function DragonDeploy() {
+  const [isOpen, setIsOpen] = useState(false)
+  const [isDeploying, setIsDeploying] = useState(false)
+  const [isProcessing, setIsProcessing] = useState(true)
+  const [files, setFiles] = useState<File[]>([])
+  const [commit, setCommit] = useState<Commit>()
+  const [deployedTxId, setDeployedTxId] = useState('')
+  const [currentDeployment, setCurrentDeployment] = useState<Deployment>()
+  const [uploadPercent, setUploadPercent] = useState(0)
+  const [currentBranch, selectedRepo, branchState, branchActions, addDeployment] = useGlobalStore((state) => [
+    state.branchState.currentBranch,
+    state.repoCoreState.selectedRepo.repo,
+    state.branchState,
+    state.branchActions,
+    state.repoCoreActions.addDeployment
+  ])
+
+  function closeModal() {
+    setIsOpen(false)
+  }
+
+  async function deploy() {
+    if (selectedRepo && commit && files.length > 0) {
+      setIsDeploying(true)
+      if (!hasIndexFile(files)) {
+        toast.error("No 'index.html' file found.")
+        setIsDeploying(false)
+        return
+      }
+
+      setUploadPercent(0)
+      try {
+        const response = await uploadFiles(files, commit, selectedRepo, setUploadPercent)
+        const deployment = await addDeployment({
+          txId: response.id,
+          commitMessage: commit.message,
+          commitOid: commit.oid
+        })
+        setCurrentDeployment(deployment)
+        setDeployedTxId(response.id)
+        toast.success('Deployed successfully')
+      } catch (err) {
+        toast.error('Deploy failed')
+      }
+      setIsDeploying(false)
+    }
+  }
+
+  useEffect(() => {
+    if (branchState.branchList.length === 0) {
+      branchActions.listBranches()
+    }
+  }, [branchState])
+
+  useEffect(() => {
+    if (selectedRepo && isOpen) {
+      ;(async () => {
+        setIsProcessing(true)
+        try {
```

Comment:
can you make a function for this logic instead of IIFE? and yeah you can also move this above with previous useEffect block.